build/Dockerfile.builder

```diff
@@ -1,4 +1,4 @@
-FROM registry.access.redhat.com/ubi9/go-toolset:1.20 as builder
+FROM registry.access.redhat.com/ubi9/go-toolset:1.21 as builder
```

Comment:
as this repo is a single repo image, https://catalog.redhat.com/software/containers/ubi9/go-toolset/61e5c00b4ec9945c18787690?architecture=amd64&image=6633d168b5bb692a240e57b3
should we use latest tag?
@rootfs , @sthaha , @vimalk78

Reply:
I'd prefer being explicit. That way the Go version used in CI doesn't drift from what we expect it to be.